gradle-plugins/brownfield/src/main/kotlin/expo/modules/plugin/ExpoBrownfieldSetupPlugin.kt

```
@@ -0,0 +1,122 @@
+import java.io.File
+
+class ExpoBrownfieldSetupPlugin : Plugin<Project> {
+  private lateinit var appProject: Project
+  private lateinit var brownfieldProject: Project
+  private lateinit var libraryExtension: LibraryExtension
+  private lateinit var appBuildDir: File
+  private lateinit var moduleBuildDir: File
+
+  override fun apply(project: Project) {
+    project.plugins.apply("maven-publish")
+    project.evaluationDependsOn(":expo")
+
+    project.afterEvaluate { project ->
+      brownfieldProject = project
+      libraryExtension = getLibraryExtension()
+      appProject = findAppProject()
+
+      appBuildDir = appProject.layout.buildDirectory.get().asFile
+      moduleBuildDir = brownfieldProject.layout.buildDirectory.get().asFile
+
+      setupSourceSets()
+      setupCopyingAutolinking()
+      setupTasks()
+      setupCopyingNativeLibs()
+    }
+  }
+
+  private fun setupSourceSets() {
+    val main = libraryExtension.sourceSets.getByName("main")
+    main.java.srcDirs("$moduleBuildDir/generated/autolinking/src/main/java")
+
+    libraryExtension.sourceSets.getByName("release").apply {
+      jniLibs.srcDirs("libsRelease")
+      assets.srcDirs("$appBuildDir/generated/assets/createBundleReleaseJsAndAssets")
+      res.srcDirs("$appBuildDir/generated/res/createBundleReleaseJsAndAssets")
+    }
+  
+    libraryExtension.sourceSets.getByName("debug").apply {
+      jniLibs.srcDirs("libsDebug")
+    }
+  }
+
+  private fun setupCopyingAutolinking() {
+    val path = "generated/autolinking/src/main/java"
+    brownfieldProject.tasks.register("copyAutolinkingSources", Copy::class.java) { task ->
+      task.dependsOn(":${appProject.name}:generateAutolinkingPackageList")
+      task.from("$appBuildDir/$path")
+      task.into("$moduleBuildDir/$path")
+
+      val rnEntryPointTask = appProject.tasks.findByName("generateReactNativeEntryPoint")
+      if (rnEntryPointTask != null) {
+        task.dependsOn(rnEntryPointTask)
+      }
```

What if the task doesn't exist? Should we throw an error?